robot/robot.py

```diff
@@ -167,7 +174,14 @@ def mode(self):
         """
         return self._game.mode
 
-    def __del__(self):
+    def close(self):
         # stop the polling threads
         for camera in self.known_cameras:
             camera._stop_poll()
+        for board_type in self.all_known_boards:
+            for board in board_type:
+                board._clean_up()
```

Comment:
I'd suggest using `del board` here. So custom deconstructors can be used on each board if necessary